crates/json-abi/src/abi.rs

```diff
@@ -608,3 +649,82 @@ impl<'de> Visitor<'de> for ContractObjectVisitor {
         Ok(ContractObject { abi, bytecode, deployed_bytecode })
     }
 }
+
+/// A [crate::Visitor] implementation that collects a mapping from struct name to set of components
+/// various structs with such name have.
```

Comment:
ah okay, I understand how this works, first we traverse the abi and collect unique names and their paths